src/OpenTelemetry/Trace/Span.cs

```diff
@@ -80,12 +96,19 @@ public sealed class Span : ISpan
             if (this.IsRecording)
             {
                 this.Activity.ActivityTraceFlags |= ActivityTraceFlags.Recorded;
-                
+
                 if (links != null)
                 {
-                    foreach (var link in links)
+                    this.links = new List<Link>(this.tracerConfiguration.MaxNumberOfLinks);
+
+                    var parentLinks = links.ToList();
+                    if (parentLinks.Count < this.tracerConfiguration.MaxNumberOfLinks)
```

Comment:
```suggestion
                    if (parentLinks.Count <= this.tracerConfiguration.MaxNumberOfLinks)
```

I guess Max is still allowed